src/diskindex.jl

```diff
@@ -159,10 +178,15 @@ splitchunks(i::AbstractArray{<:CartesianIndex}, chunks) =
 splitchunks(i::AbstractArray{Bool}, chunks) = splitchunks(size(i), (), chunks)
 splitchunks(i::CartesianIndices, chunks) = splitchunks(i.indices, (), chunks)
 splitchunks(i::CartesianIndex, chunks) = splitchunks(i.I, (), chunks)
-splitchunks(_, chunks) = (first(chunks),), Base.tail(chunks)
+splitchunks(_, chunks) = (first(chunks),), tail(chunks)
 splitchunks(si, chunksnow, chunksrem) =
-    splitchunks(Base.tail(si), (chunksnow..., first(chunksrem)), Base.tail(chunksrem))
+    splitchunks(tail(si), (chunksnow..., first(chunksrem)), tail(chunksrem))
+function splitchunks(si, chunksnow, chunksrem::Tuple{})
+    checktrailing(si[1])
+    splitchunks(tail(si), chunksnow, chunksrem)
+end
 splitchunks(::Tuple{}, chunksnow, chunksrem) = (chunksnow, chunksrem)
+splitchunks(::Tuple{}, chunksnow, chunksrem::Tuple{}) = (chunksnow, chunksrem)
```

Comment:
I am not sure why modifications here are necessary. Could you maybe try again without this new methods. I have a slight suspicion this is only necessary because the implementation of `process_index` above is not correct.

Reply:
Thanks for reverting, I will investigate from here